frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTab.tsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+import {
+  Button,
+  Tooltip,
+  TextListItem,
+  TextListItemVariants,
+  TextContent,
+  Spinner,
+  Stack,
+  StackItem,
+  Split,
+  SplitItem,
+  Bullseye,
+} from '@patternfly/react-core';
+import OutlinedPlayCircleIcon from '@patternfly/react-icons/dist/esm/icons/outlined-play-circle-icon';
+import PauseIcon from '@patternfly/react-icons/dist/esm/icons/pause-icon';
+import PlayIcon from '@patternfly/react-icons/dist/esm/icons/play-icon';
+import DownloadIcon from '@patternfly/react-icons/dist/esm/icons/download-icon';
+import { PipelineRunTaskDetails } from '~/concepts/pipelines/content/types';
+import SimpleDropdownSelect from '~/components/SimpleDropdownSelect';
+import useFetchLogs from '~/concepts/k8s/pods/useFetchLogs';
+import usePodContainerState from '~/concepts/k8s/pods/usePodContainerState';
+import LogsTabStatus from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/LogsTabStatus';
+import { LOG_TAIL_LINES } from '~/concepts/pipelines/content/pipelinesDetails/pipelineRun/runLogs/const';
+import DashboardCodeEditor from '~/concepts/dashboard/codeEditor/DashboardCodeEditor';
+import useCodeEditorAsLogs from '~/concepts/dashboard/codeEditor/useCodeEditorAsLogs';
+import { downloadFullPodLog } from '~/concepts/k8s/pods/utils';
+import { usePipelinesAPI } from '~/concepts/pipelines/context';
+
+type LogsTabProps = {
+  task: PipelineRunTaskDetails;
+};
+
+const NO_POD_ERROR = new Error('No pod found, unable to show logs');
+
+const LogsTab: React.FC<LogsTabProps> = ({ task }) => {
+  const { namespace } = usePipelinesAPI();
+  const podName = task.runDetails?.status.podName ?? '';
+  const podNameError = !podName ? NO_POD_ERROR : undefined;
+  const { podLoaded, podError, podContainers, selectedContainer, setSelectedContainer } =
```

Comment:
I need to dig deeper on the state rabbit hole but I'll left the edge case in case you want to take a look too @andrewballantyne 
If you create a run, open the LogsTab and keep the tab open until the pod is deleted (in case it fails) it won't trigger the error, but it will keep a log message (see gif), you'll get the error if you close and open the tab again.
![logs](https://github.com/opendatahub-io/odh-dashboard/assets/16117276/01c548c4-d7be-4a09-a368-4c2737bf2251)

Reply:
Your concern is what? Having stale data? I'm not sure this is a problem 🤔 We don't clean up the UI on the fly for anything else. With the looping of the gif, perhaps I'm missing the issue?

Reply:
That we have two different screens for the same state (which is that the pod has failed). If we get the log tab open it will display this:
![image](https://github.com/opendatahub-io/odh-dashboard/assets/14068621/c840b8a9-8f7d-44b3-b284-5f4fe7b600e5)
But if we close and open again we'll get the intended error message:
<img width="1010" alt="Screenshot 2023-08-31 at 18 21 57" src="https://github.com/opendatahub-io/odh-dashboard/assets/16117276/7a257516-2605-43cf-a128-c813ea973031">

To be honest, not a big issue for me, it's just something that I think we can improve overtime.

Reply:
Okay, so you're saying if the pod doesn't have proper state, the message we show is misleading as there is no real pod behind it? Okay, noted -- I think we will have to deal with that later.

Closing and reopening seeing a different state is _not_ a problem? 🤔

Reply:
I think what we have now is acceptable from a UX perspective. But can we detect and show the error state in the subsequent auto-loading rather than relying on a manual refresh (closing and reopening the side drawer in this case)? 

BTW, I don't know how to execute the code so I'm only judging from the screenshot. Just forget it if we've already done that.

Reply:
> Okay, so you're saying if the pod doesn't have proper state, the message we show is misleading as there is no real pod behind it? Okay, noted -- I think we will have to deal with that later.
Yeah, that was basically my main complaint there, reopening it was just a nit but I don't think is a problem